packages/headless/src/features/search-parameters/search-parameter-analytics-actions.ts

```diff
@@ -0,0 +1,34 @@
+import {createAsyncThunk} from '@reduxjs/toolkit';
+import {AsyncThunkSearchOptions} from '../../api/search/search-api-client';
+import {SearchParameters} from '../../controllers';
+import {SearchParametersState} from '../../state/search-app-state';
+import {ConfigurationSection} from '../../state/state-sections';
+import {AnalyticsType} from '../analytics/analytics-utils';
+import {logSearchboxSubmit} from '../query/query-analytics-actions';
+import {logResultsSort} from '../sort-criteria/sort-criteria-analytics-actions';
+
+/**
+ * Logs the analytics event that corresponds best with the difference between previous and new search parameters
+ * @param previousParameters The search parameters before the update
+ * @param newParameters The search parameters after the update
+ */
+export const logSearchParametersChange = createAsyncThunk<
```

Comment:
Not sure about this...

I added a separate action thunk here, I could've done those comparisons in the controller as well and pick the right action there.

Advantages of creating an action is we can log multiple analytics if multiple params would change at once.
Although I don't think UA supports having multiple search cause for a single search event, would they support/like having multiple UA calls for a single search? @olamothe ?
The JSUI only logged the most relevant `query expression > sort > facet`

Reply:
Hmm yeah Duplicate Search ID is an incoherent event, so better not implement multiple calls, should I remove that option and just compare changes & pick the event in the controller directly? 🤔

Reply:
I think we should either remove it or at least not expose it, because it is too complex use on it's own. It needs one to pass two sets of parameters, and remember to merge with the initial state.

Reply:
Good! I'll remove